### PR TITLE
Properly place and print required vs. non-required args in the help menu.  Addresses #219.

### PR DESCRIPTION
**Pull request by DailyDreaming**
Addresses #219.

This PR separates the args into `Positional Arguments`, `Required Arguments`, and `Optional Arguments` based on whether the swagger (or internal CLI function) lists them as required or not.

So this only groups arguments differently.

For example, currently it prints:
```
(v3nv) quokka@qcore:~/git/dcp-cli$ hca dss get-file -h
usage: hca dss get-file [-h] --uuid UUID --replica {aws,gcp}
                        [--version VERSION] [--token TOKEN]

Given a file UUID, return the latest version of that file. If the version is
provided, that version of the file is returned instead. Headers will contain
the data store metadata for the file. This endpoint returns a HTTP redirect to
another HTTP endpoint with the file contents.

optional arguments:
  -h, --help           show this help message and exit
  --version VERSION    Timestamp of file creation in DSS_VERSION format. If
                       this is not provided, the latest version is returned.
  --token TOKEN        Token to manage retries. End users constructing queries
                       should not set this parameter.
  --uuid UUID          A RFC4122-compliant ID for the file.
  --replica {aws,gcp}  Replica to fetch from.
```
This would change to:
```
(v3nv) quokka@qcore:~/git/dcp-cli$ hca dss get-file -h
usage: hca dss get-file [-h] --uuid UUID --replica {aws,gcp}
                        [--version VERSION] [--token TOKEN]

Given a file UUID, return the latest version of that file. If the version is
provided, that version of the file is returned instead. Headers will contain
the data store metadata for the file. This endpoint returns a HTTP redirect to
another HTTP endpoint with the file contents.

Optional Arguments:
  -h, --help           show this help message and exit
  --version VERSION    Timestamp of file creation in DSS_VERSION format. If
                       this is not provided, the latest version is returned.
  --token TOKEN        Token to manage retries. End users constructing queries
                       should not set this parameter.

Required Arguments:
  --uuid UUID          A RFC4122-compliant ID for the file.
  --replica {aws,gcp}  Replica to fetch from.
```

### CODE REVIEW

File: hca/util/__init__.py [hunk 1]
```diff
@@ -83,9 +83,15 @@ def special_cli_command(self, required_argument, optional_argument=""):
 
 from __future__ import absolute_import, division, print_function, unicode_literals
 
-import os, types, collections, typing, json, errno, base64, argparse
+import os
```
Comment (Bento007) [inline hca/util/__init__.py]: Why?

Comment (DailyDreaming) [reply hca/util/__init__.py]: PEP8: https://www.python.org/dev/peps/pep-0008/#imports

**Comment by DailyDreaming** [comment]
@Bento007 Should be addressed.  Example output now copy-pasted at the top.

File: hca/util/__init__.py [hunk 2]
```diff
@@ -520,13 +526,15 @@ def _get_param_argparse_type(self, anno):
     def build_argparse_subparsers(self, subparsers):
         for method_name, method_data in self.methods.items():
             subcommand_name = method_name.replace("_", "-")
-            subparser = subparsers.add_parser(subcommand_name, help=method_data.get("summary"),
-                                              description=method_data.get("description"))
+            optional_args = subparsers.add_parser(subcommand_name, help=method_data.get("summary"),
+                                                  description=method_data.get("description"))
+            required_args = optional_args.add_argument_group('Required Arguments')
```
[the remark this answers is not in the document]
Comment (DailyDreaming) [reply hca/util/__init__.py]: Ah, good point.  I can always use better naming, thanks!

File: .gitignore [hunk 3]
```diff
@@ -42,3 +42,7 @@ npm-debug.log
 
 # Credential files
 /gcs-credentials.json
+
+# virtualenv
+venv/
```
Comment (Bento007) [inline .gitignore]: this should be in your global .gitignore not in this repo.

Comment (DailyDreaming) [reply .gitignore]: How do you distinguish the two?

**Pull request merged**